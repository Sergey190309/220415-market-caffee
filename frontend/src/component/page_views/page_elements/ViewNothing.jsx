import React from 'react'
import PropTypes from 'prop-types'

const ViewNothing = props => {
  return (
    <div>ViewNothing</div>
  )
}
ViewNothing.defaultProps = {}
ViewNothing.propTypes = {}

export default ViewNothing