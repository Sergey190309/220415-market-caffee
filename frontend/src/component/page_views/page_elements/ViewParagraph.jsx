import React from 'react'
import { useAppState, useAppEffect, useAppContext } from '../../../hooks/react'
import { useAppSelector } from '../../../hooks/reactRedux'
import PropTypes from 'prop-types'

import { deviceSelector } from '../../../redux/slices'
import { CONTENT_REQUESTED } from '../../../redux/constants/types'
import { useSaga } from '../../../redux/contentSaga/createIO'
import { getContentSaga } from '../../../redux/contentSaga/content'
import { LandingContext } from '../../../context'
import { Box, Divider, Tooltip, Typography } from '@mui/material'
import * as SZ from '../../../constants/sizes'
import * as CL from '../../../constants/colors'

const ViewParagraph = ({ recordId, initialState }) => {
  /**
  * States:
  * state: object - Paragraph content loaded from back-end
  *    using getContentSaga.
  * content: object - Content itself shown on the component.
  *    Updated with useEffect.
  * changed: boolean - Indication content was changed from last
  *    download or upload. Updated with useEffect.
  * contextMenuOpened: boolean - Self explain. Open with right
  *    button.
  * upperLevelContextMenuOpened: boolean. Open when appropriate button
  *    pressed on context menu.
  * paragraphEditted: boolean - Set close showing, open
  *    textboxes for edition.
  * indicatorOpened: boolean - Set component indicator on and
  *    off respectevily.
  */
  const [content, setContent] = useAppState(initialState)
  const [changed, setChanged] = useAppState(false)
  const [tooltipVisible, setTooltipVisible] = useAppState(false)

  const [state, getSagaDispatch] = useSaga(getContentSaga, initialState)

  const { editable } = useAppSelector(deviceSelector)

  const { componentName } = useAppContext(LandingContext)
  const { componentName: viewName } = useAppContext(LandingContext)

  useAppEffect(() => { // Saga
    // console.log('ViewParagraph:',
    //   '\n useEffect[recordId, kind]',
    //   '\n  recordId ->', recordId)
    // indexRef.current = +recordId.split('_').pop()
    // if (kind === '') {
      setChanged(false)
      getSagaDispatch({
        type: CONTENT_REQUESTED,
        payload: {
          identity: recordId,
          view_id: viewName
        }
      })
    // setContent(state)
    // console.log('ViewParagraph, state ->', state)
    // }
    // }, [recordId, kind])
  }, [])

  useAppEffect(() => {
    setContent(state)
  }, [state])


  const NormalOutput = () => (
    <Tooltip
      title={recordId} placement='top' followCursor arrow
      open={tooltipVisible}
      onOpen={() => { setTooltipVisible(editable && true) }}
      onClose={() => { setTooltipVisible(false) }}
    >
      <Box
        sx={{
          border: SZ.blockBorder, borderColor: 'text.disabled', borderRadius: 3,
          m: '.25rem',
          p: '.5rem',
          '&:hover': editable && {
            borderColor: CL.attention
          }
        }}
      >
        {/* Fuck! */}
        <Typography variant='h6'>
          {content.title}
        </Typography>
        {content.title && content.content.length > 0
          ? <Divider />
          : null}
        {content.content.map((item, index) => (
          <Typography key={index} variant='body2'>
            {item}
          </Typography>
        ))}
      </Box>
    </Tooltip>
  )

  return (
    <NormalOutput />
  )
}
ViewParagraph.defaultProps = {
  recordId: '',
  initialState: {
    title: '', // info from back-end
    content: ['']
  }
}
ViewParagraph.propTypes = {
  recordId: PropTypes.string.isRequired,
  initialState: PropTypes.object.isRequired
}

export default ViewParagraph