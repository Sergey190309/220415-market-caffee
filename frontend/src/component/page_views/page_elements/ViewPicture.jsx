import React from 'react'
import PropTypes from 'prop-types'

const ViewPicture = props => {
  return (
    <div>ViewPicture</div>
  )
}
ViewPicture.defaultProps = {}
ViewPicture.propTypes = {}

export default ViewPicture