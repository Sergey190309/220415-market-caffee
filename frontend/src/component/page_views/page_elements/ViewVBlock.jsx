import React from 'react'
import PropTypes from 'prop-types'

const ViewVBlock = props => {
  return (
    <div>ViewVBlock</div>
  )
}
ViewVBlock.defaultProps = {}
ViewVBlock.propTypes = {}

export default ViewVBlock