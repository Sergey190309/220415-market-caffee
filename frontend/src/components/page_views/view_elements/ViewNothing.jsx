import React from 'react'
import PropTypes from 'prop-types'

const ViewNothing = ({ recordId, viewName, lng }) => {
  // console.log('ViewNothing, props are', recordId, viewName, lng )
  return (
    <div>
      <h1>ViewNothing</h1>
    </div>
  )
}

ViewNothing.propTypes = {
  recordId: PropTypes.string.isRequired,
  viewName: PropTypes.string.isRequired,
  lng: PropTypes.string.isRequired
}

export default ViewNothing
