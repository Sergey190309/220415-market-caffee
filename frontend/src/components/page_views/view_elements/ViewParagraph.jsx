import React, { useState, useEffect, useRef, Fragment } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import PropTypes from 'prop-types'
import { Message, Divider } from 'semantic-ui-react'

import {
  CONTENT_REQUESTED
} from '../../../redux/constants/types'
import { useSaga } from '../../../redux/saga/content/createIO'
import {
  getContentSaga
  // putContentSaga
} from '../../../redux/saga/content/content'
import {
  deviceSelector,
  backendUpdateSelector,
  backendTxtUpdateReady, backendTxtUpdateStart,
  resetBackendUpdate
} from '../../../redux/slices'
import { createContextFromEvent } from './editors/createContextFromEvent' // tested
import ParagraphContextMenu
  from './editors/ParagraphContextMenu' // tested
import SaveToBackendContextMenu from '../../items/SaveToBackendContextMenu'
import ParagraphEditor from './editors/ParagraphEditor' // tested
import Indicator from './indicator/Indicator'

const ViewParagraph = ({
  initialState, recordId, viewName,
  addElementProp, deleteElementProp
  // upperLvlAddElementProp, upperLvlDeleteElementProp
}) => {
  /**
   * States:
   * state: object - Paragraph content loaded from back-end
   *    using getContentSaga.
   * content: object - Content itself shown on the component.
   *    Updated with useEffect.
   * changed: boolean - Indication content was changed from last
   *    download or upload. Updated with useEffect.
   * contextMenuOpened: boolean - Self explain. Open with right
   *    button.
   * paragraphEditted: boolean - Set close showing, open
   *    textboxes for edition.
   * indicatorOpened: boolean - Set component indicator on and
   *    off respectevily.
   * Store state variables.
   * editable: boolean - Admin only can set this variable on
   *    admin page.
   * loaded: boolean - Indication whether changed content
   *    successfully uploaded to back-end. Used to set save to
   *    back-end context menu disabled.
   */
  const [state, getSagaDispatch] = useSaga(
    getContentSaga, initialState)
  const [content, setContent] = useState({
    title: '',
    content: ['']
  })
  const [changed, setChanged] = useState(false)
  const [contextMenuOpened, setContextMenuOpened] = useState(false)
  const [paragraphEditted, setParagraphEditted] = useState(false)
  const [indicatorOpened, setIndicatorOpened] = useState(false)
  const { editable } = useSelector(deviceSelector)
  const { loaded, kind } = useSelector(backendUpdateSelector)
  const dispatch = useDispatch()

  const contextRef = useRef(null)
  const indicatorRef = useRef(null)

  useEffect(() => { // Saga
    // console.log('ViewParagraph, useEffect(getSagaDispatch), recordId ->', recordId)
    getSagaDispatch({
      type: CONTENT_REQUESTED,
      payload: {
        identity: recordId,
        view_id: viewName
      }
    })
  }, [recordId, viewName])

  useEffect(() => {
    setContent(state)
  }, [state])

  useEffect(() => {
    if (JSON.stringify(state) !== JSON.stringify(content)) {
      // console.log('ViewParagraph:\n useEffect (content)',
      //   '\n  state ->', state,
      //   '\n  content', content
      // )
      dispatch(backendTxtUpdateReady({
        identity: recordId,
        view_id: viewName,
        content: content
      }))
      setChanged(true)
    }
  }, [content])

  useEffect(() => {
    if (changed) {
      // console.log('componetns, ViewParagraph, useEffect ->')
      setChanged(false)
      dispatch(resetBackendUpdate())
    }
  }, [loaded])

  const onClickHandler = event => {
    // console.log('components, page_view, view_elements, ViewParagraph, identities ->', identities)
    event.preventDefault()
    if (editable) {
      setIndicatorOpened(editable && !indicatorOpened)
      indicatorRef.current = createContextFromEvent(event)
    }
  }

  const onContextMenuHendler = event => {
    event.preventDefault()
    contextRef.current = createContextFromEvent(event)
    setIndicatorOpened(false)
    setContextMenuOpened(true)
  }

  const saveToBackend = () => {
    // console.log('ViewParagraph, saveToBackend, content ->', content)
    dispatch(backendTxtUpdateStart())
  }
  const addAbove = () => {
    /**
     * To send signal one module above to change structure
     */
    const recordIndex = parseInt(recordId.split('_').pop())
    if (isNaN(recordIndex)) {
      console.log('Unable to parse recordId for index')
    } else { addElementProp(recordIndex) }
  }
  const addBelow = () => {
    /**
     * To send signal one module above to change structure
     */
    const recordIndex = parseInt(recordId.split('_').pop())
    if (isNaN(recordIndex)) {
      console.log('Unable to parse recordId for index')
    } else { addElementProp(recordIndex + 1) }
  }
  const deleteElement = () => {
    /**
     * To send signal one module above to change structure
     */
    const recordIndex = parseInt(recordId.split('_').pop())
    // console.log('ViewParagraph, deleteFmBackend')
    if (isNaN(recordIndex)) {
      console.log('Unable to parse recordId for index')
    } else { deleteElementProp(recordIndex) }
  }

  const normalOutput = () => (
    <Message
      onContextMenu={editable
        ? onContextMenuHendler
        : null
      }
      onClick={onClickHandler}
      data-testid='Message'
    >
      <Message.Header content={content.title} />
      {content.title && content.content.length > 0
        ? <Divider />
        : null}
      {content.content.map((item, index) => (
        <Message.Item as='p' key={index}>
          {item}
        </Message.Item>
      ))}
    </Message>
  )

  const paragraphEditorConf = () => (
    <ParagraphEditor
      setParagraphEditted={setParagraphEditted}
      comingContent={content}
      setComingContent={setContent}
    />
  )

  const indicatorConf = () => (
    <Indicator
      isOpened={indicatorOpened}
      context={indicatorRef}
      header={viewName}
      content={recordId}
      setIndicatorOpened={setIndicatorOpened}
    />
  )

  const paragraphContextMenuConf = () => (
    <ParagraphContextMenu
      isOpened={contextMenuOpened}
      saveDisabled={!changed}
      context={contextRef}
      setContextMenuOpened={setContextMenuOpened}
      setParagraphEditted={setParagraphEditted}
      saveToBackend={saveToBackend}
      deleteElement={deleteElement}
      addAbove={addAbove}
      addBelow={addBelow}
    />
  )

  return (
    <Fragment>
      {editable
        ? paragraphEditted
          ? paragraphEditorConf()
          : <Fragment>
              {normalOutput()}
              {indicatorOpened
                ? indicatorConf()
                : null
              }
              {contextMenuOpened
                ? (kind === '')
                    ? paragraphContextMenuConf()
                    : changed
                      ? paragraphContextMenuConf()
                      : <SaveToBackendContextMenu />
                : null
              }
            </Fragment>
        : normalOutput()
      }
      {/* {editable
        ? paragraphEditted
          ? <ParagraphEditor
            setParagraphEditted={setParagraphEditted}
            comingContent={content}
            setComingContent={setContent}
          />
          : <ParagraphContextMenu
            isOpened={contextMenuOpened}
            // saveDisabled={false}
            saveDisabled={!changed}
            context={contextRef}
            setContextMenuOpened={setContextMenuOpened}
            setParagraphEditted={setParagraphEditted}
            saveToBackend={saveToBackend}
            deleteElement={deleteElement}
            addAbove={addAbove}
            addBelow={addBelow}
          // upperLvlAddElementProp={upperLvlAddElementProp}
          // upperLvlDeleteElementProp={upperLvlDeleteElementProp}
          />
        : null
      }
      {paragraphEditted
        ? null
        : <NormalOutput />
      }
      {indicatorOpened
        ? <Indicator
          isOpened={indicatorOpened}
          context={indicatorRef}
          header={viewName}
          content={recordId}
          setIndicatorOpened={setIndicatorOpened}
        />
        : null
      } */}
    </Fragment>
  )
}

ViewParagraph.defaultProps = {
  initialState: {
    title: '', // info from back-end
    content: ['']
  },
  recordId: '',
  viewName: '',
  addElementProp: () => { },
  deleteElementProp: () => { }
  // upperLvlAddElementProp: () => { },
  // upperLvlDeleteElementProp: () => {}
}

ViewParagraph.propTypes = {
  initialState: PropTypes.object.isRequired,
  recordId: PropTypes.string.isRequired,
  viewName: PropTypes.string.isRequired,
  addElementProp: PropTypes.func.isRequired,
  deleteElementProp: PropTypes.func.isRequired
  // upperLvlAddElementProp: PropTypes.func.isRequired,
  // upperLvlDeleteElementProp: PropTypes.func.isRequired
}

export default ViewParagraph
