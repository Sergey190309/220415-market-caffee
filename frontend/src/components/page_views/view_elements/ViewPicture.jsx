import React, { Fragment, useEffect } from 'react'
import { Image } from 'semantic-ui-react'
import PropTypes from 'prop-types'

import { PICTURE_REQUESTED, CONTENT_REQUESTED } from '../../../redux/constants/types'
import { useSaga } from '../../../redux/saga/contentLoading/createIO'
import { contentSaga, picSaga } from '../../../redux/saga/contentLoading/contentLoading'

const ViewPicture = ({ recordId, viewName, lng, dimension }) => {
  /**
   * dimention {
   *  direction: [vertical | horizontal]
   *  suze: in pixels
   * }
   */
  const [textState, textSagaDispatch] = useSaga(contentSaga, {
    title: '',
    content: ''
  })
  const [picState, picSagaDispatch] = useSaga(picSaga, { pic: '' })

  useEffect(() => {
    const payload = {
      identity: recordId,
      view_id: viewName
      // locale_id: lng,
    }
    picSagaDispatch({
      type: PICTURE_REQUESTED,
      payload
    })
    payload.locale_id = lng
    textSagaDispatch({
      type: CONTENT_REQUESTED,
      payload
    })
  }, [recordId, viewName, lng, textSagaDispatch, picSagaDispatch])
  // console.log('ViewPictures, picState ->', picState)
  return (
    <Fragment>
      <h4>{textState.title}</h4>
      <Image
        src={`data:image;charset=utf-8;base64,${picState.pic}`}
        spaced
        height={dimension.direction === 'vertical' ? dimension.size : null}
        width={dimension.direction === 'horizontal' ? dimension.size : null}
        // centered
        // size='small'
        // wrapped
        // bordered
        // fluid
        />
        <p>{textState.content}</p>
    </Fragment>
  )
}

ViewPicture.defaultProps = {
  recordId: '',
  viewName: '',
  lng: '',
  dimension: {}
}

ViewPicture.propTypes = {
  recordId: PropTypes.string.isRequired,
  viewName: PropTypes.string.isRequired,
  lng: PropTypes.string.isRequired,
  dimension: PropTypes.object.isRequired
}

export default ViewPicture
