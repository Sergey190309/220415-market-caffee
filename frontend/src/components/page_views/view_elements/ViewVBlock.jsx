import React, { useState, useEffect, Fragment } from 'react'
// import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import { Container, Divider } from 'semantic-ui-react'

import { makeRecordIdList } from '../../../utils/utils'

import ViewParagraph from './ViewParagraph'
import ViewPicture from './ViewPicture'
import ViewNothing from './ViewNothing'

const ViewVBlock = ({ recordsId, viewName, lng }) => {
  /**
   * recordsId structure - 01_vblock_txt_3
   * 01 - serial number
   * vblock - kind of structure
   * txt - kind of structure elements
   * 3 - element qnt
   */
  const [recordIdList, setRecordIdList] = useState([])
  /**
   * recordIdList - list of identities in content table.
   */

  useEffect(() => {
    setRecordIdList(makeRecordIdList(recordsId))
  }, [recordsId])

  const props = {
    viewName: viewName,
    lng: lng
  }
  // console.log('ViewVBlock, recordIdList ->', recordIdList);

  if (recordsId.includes('txt')) {
    return recordIdList.map(txtRecordId => {
      // console.log('ViewVBlock output, recordId ->', recordId)
      return (
        <Fragment key={txtRecordId}>
          <ViewParagraph {...props} recordId={txtRecordId} />
        </Fragment>
      )
    })
  }
  if (recordsId.includes('pix')) {
    // console.log('ViewHBlock, props ->', props)
    return recordIdList.map((pixRecordId, index) => {
      return (
        <Container textAlign='center' key={pixRecordId} >
          <ViewPicture
            {...props}
            recordId={pixRecordId}
            dimension={{ direction: 'horizontal', size: 250 }}
          />
          {index < recordIdList.length - 1 ? <Divider /> : null}
        </Container>
      )
    })
  }
  return <ViewNothing />
}

ViewVBlock.defaultProps = {
  recordsId: '',
  viewName: '',
  lng: ''
}

ViewVBlock.propTypes = {
  recordsId: PropTypes.string.isRequired,
  viewName: PropTypes.string.isRequired,
  lng: PropTypes.string.isRequired
}

// const mapStateToProps = state => ({
//   lng: state.lng,
// });

// const mapDispatchToProps = dispatch => ({
// structureStart: viewName => dispatch(structureStart(viewName)),
// });

export default ViewVBlock
// export default connect(mapStateToProps)(ViewVBlock);
