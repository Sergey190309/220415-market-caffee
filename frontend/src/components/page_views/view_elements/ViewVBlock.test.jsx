import React from 'react';
import { render, screen } from '@testing-library/react';

import { makeRecordIdList } from '../../../utils/utils';

import ViewVBlock from './ViewVBlock';

// import { output } from './ViewVBlock';

import ViewParagraph from './ViewParagraph';
import ViewPicture from './ViewPicture';
import ViewNothing from './ViewNothing';

const mockViewParagraph = ({ recordId, viewName, lng }) => {
  return <div data-testid='ViewParagraph' />;
};
const mockViewPicture = ({ recordId, viewName, lng }) => {
  return <div data-testid='ViewPicture' />;
};
const mockViewNothing = ({ recordId, viewName, lng }) => {
  return <div data-testid='ViewNothing' />;
};

jest.mock('./ViewParagraph', () => ({ __esModule: true, default: jest.fn() }));
jest.mock('./ViewPicture', () => ({ __esModule: true, default: jest.fn() }));
jest.mock('./ViewNothing', () => ({ __esModule: true, default: jest.fn() }));

describe('ViewVBlock testing', () => {
  beforeEach(() => {
    ViewParagraph.mockImplementation(mockViewParagraph);
    ViewPicture.mockImplementation(mockViewPicture);
    ViewNothing.mockImplementation(mockViewNothing);
  });

  test('output calling ViewParagraph', () => {
    const testProps = {
      recordsId: '01_vblock_txt_3',
      viewName: 'mockViewName',
      lng: 'mockLng',
    };
    const mockRecordIdList = makeRecordIdList(testProps['recordsId']);
      const viewParagraphPropsList = mockRecordIdList.map(item => ({
      viewName: testProps['viewName'],
      lng: testProps['lng'],
      recordId: item,
    }));
    render(<ViewVBlock {...testProps} />);
    expect(ViewParagraph).toHaveBeenCalledTimes(3);
    expect(ViewPicture).toHaveBeenCalledTimes(0);
    expect(ViewNothing).toHaveBeenCalledTimes(0);
    viewParagraphPropsList.forEach((item, index) => {
      expect(ViewParagraph.mock.calls[index][0]).toEqual(item);
    });
    // console.log('output calling ViewParagraph ->', ViewParagraph.mock.calls);
  });
  test('output calling ViewPicture', () => {
    /**
     * last failed:
     * I've chanched arguments.
     */
    const testProps = {
      recordsId: '01_vblock_pix_4',
      viewName: 'mockViewName',
      lng: 'mockLng',
    };
    const mockRecordIdList = makeRecordIdList(testProps['recordsId']);
      const viewParagraphPropsList = mockRecordIdList.map(item => ({
        viewName: testProps['viewName'],
        lng: testProps['lng'],
        recordId: item,
        dimension: { direction: 'horizontal', size: 250 }
      }));
    render(<ViewVBlock {...testProps} />);
    expect(ViewParagraph).toHaveBeenCalledTimes(0);
    expect(ViewPicture).toHaveBeenCalledTimes(4);
    expect(ViewNothing).toHaveBeenCalledTimes(0);
    viewParagraphPropsList.forEach((item, index) => {
      expect(ViewPicture.mock.calls[index][0]).toEqual(item);
      // console.log('output calling ViewPicture  ->', ViewPicture.mock.calls[index][0]);
    });
  });
  test('output calling ViewNothing', () => {
    const testProps = {
      recordId: '01_vblock_xxx_4',
      viewName: 'mockViewName',
      lng: 'mockLng',
    };
    render(<ViewVBlock {...testProps} />);
    expect(ViewParagraph).toHaveBeenCalledTimes(0);
    expect(ViewPicture).toHaveBeenCalledTimes(0);
    expect(ViewNothing).toHaveBeenCalledTimes(2);
    // console.log('output calling ViewNothing ->', ViewNothing.mock.calls);
  });

});
