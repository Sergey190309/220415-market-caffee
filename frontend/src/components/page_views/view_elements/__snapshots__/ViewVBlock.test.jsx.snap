// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewVBlock testing appearance rendering, nothing 1`] = `
<div
  class="ui container"
  data-testid="Output"
>
  <div
    data-textid="mockViewNothing"
  />
</div>
`;

exports[`ViewVBlock testing appearance rendering, pix 1`] = `
<div
  class="ui container"
  data-testid="Output"
>
  <div
    class="ui center aligned container"
  >
    <div
      data-textid="mockViewPicture"
    />
    <div
      class="ui divider"
    />
  </div>
  <div
    class="ui center aligned container"
  >
    <div
      data-textid="mockViewPicture"
    />
    <div
      class="ui divider"
    />
  </div>
  <div
    class="ui center aligned container"
  >
    <div
      data-textid="mockViewPicture"
    />
  </div>
</div>
`;

exports[`ViewVBlock testing appearance rendering, txt 1`] = `
<div
  class="ui container"
  data-testid="Output"
>
  <div
    data-textid="mockViewParagraph"
  />
  <div
    data-textid="mockViewParagraph"
  />
  <div
    data-textid="mockViewParagraph"
  />
  <div
    data-textid="mockViewParagraph"
  />
</div>
`;
